controllers/topology/internal/mergepatch/mergepatch_test.go

```diff
@@ -184,6 +212,35 @@ func TestNewHelper(t *testing.T) {
 			wantHasSpecChanges: true,
 			wantPatch:          []byte("{\"spec\":{\"map\":{\"A\":\"A\",\"C\":\"C\"}}}"),
 		},
+		{
+			name: "Value of type map, enforces entries from modified if the path is authoritative",
+			original: &unstructured.Unstructured{
+				Object: map[string]interface{}{
+					"spec": map[string]interface{}{
+						"map": map[string]string{
+							"A": "A-changed",
+							"B": "B",
+							// C missing
+						},
+					},
+				},
+			},
+			modified: &unstructured.Unstructured{
+				Object: map[string]interface{}{
+					"spec": map[string]interface{}{
+						"map": map[string]string{
+							"A": "A",
+							// B missing
+							"C": "C",
+						},
+					},
+				},
+			},
+			options:            []HelperOption{AuthoritativePaths{contract.Path{"spec", "map"}}},
+			wantHasChanges:     true,
+			wantHasSpecChanges: true,
+			wantPatch:          []byte("{\"spec\":{\"map\":{\"A\":\"A\",\"B\":null,\"C\":\"C\"}}}"),
```

Comment:
Maybe add a test case to show that ignorePaths superseded the AuthoritativePath to document that behaviour.